### ISSUE TITLE
Remittances & % of GDP preprocessor series

### ISSUE THREAD

**Issue by quciet** [PardeeCenterDU/IFs-Issues-Tracking#83]
Two issues.

1. Data issue

Some preprocessors in IFs, which ought to be in constant dollars, were mistakenly brought in as current dollars. For example, ImportServices, ImportsMerchandise, XWorkerRemitReceived, etc. Those series are in WDI, and WDI now provides them as % of GDP values. 

Thus, the data team created some new data series- ExportServices%GDP, ExportsMerchandise%GDP, ImportServices%GDP, ImportsMerchandise%GDP, XWorkerRemitPaid%GDP, XWorkerRemitReceived%GDP. To use them in the models, I believe there will be some change of code to initialize data based on % of GDP instead of actual dollar values.

2. Remittance & Migration

The remittances data we are using (XWorkerRemitReceived & XWorkerRemitPaid which will be replaced by XWorkerRemitPaid%GDP, XWorkerRemitReceived%GDP according to the point above) contains both compensation and personal transfer. However, to better estimate migration, compensation should be excluded (correct me if I'm wrong here, Dave). 

We found a data source from IMF which separates compensation & personal transfer. Those data are in current US dollars I believe. So we need to think about how those data can be used.

Please comment below on anything I missed here, thank you!!

**Comment by PardeeCenterIFs** [comment]
I assume this means we need to convert that data on comensation & personal transfer to constant dollars, is that something the data team can do?

**Comment by quciet** [comment]
For the 1st point above, José will make changes in the code to fix the Data issue.

The 2nd point which is about separating Remittance data has been moved to #96 here and will be handled by the data team later with a lower priority.

Thanks much!

**Comment by PardeeCenterIFs** [comment]
Implementation Completed. Testing to be done. I will send the file to test in Development environment.

**Comment by quciet** [comment]
After a quick checking, I don't see big issues. However, there is one thing I'm not quite understanding.

For this data change, we updated the sectoral trade data (services%GDP & goods% GDP) which caused some big changes in the initialization for a few countries.

![image](https://user-images.githubusercontent.com/34465524/141861556-72f372bb-6458-494d-8327-13db4d6ba1f7.png)

However, we did not touch the total trade data (GoodSer%GDP). I was expecting that the initialization of the total trade (import & export) won't change. But the total trade changed a bit- 
![image](https://user-images.githubusercontent.com/34465524/141861385-206b61d8-58db-417c-9b94-9e7113399b78.png)

So I'm not sure how the model reconciles these numbers.

**Comment by PardeeCenterIFs** [comment]
In the first year of the run the model computes totals based on the sectoral values:

            '   First Year:  Trade
            '   ------------------

            For R% = 1 To NR%
                X(R%) = 0
                M(R%) = 0
                For S% = 1 To NSec%
                    X(R%) = X(R%) + XS(R%, S%)
                    M(R%) = M(R%) + MS(R%, S%)
                Next S%
            Next R%

**Comment by quciet** [comment]
Thank you! I'm moving this to a lower priority because the change can be implemented into IFs now.

I'm still curious- 
You said the total value was calculated based on the sectoral values, but I also see a preprocessor series that initialize the total trade. So what the relationship between these series, namely

SeriesImportGoodSer%GDP
ImportServices%GDP 
ImportsMerchandise%GDP

![image](https://user-images.githubusercontent.com/34465524/142480278-850274f4-e09a-4341-9c89-f8a54895f902.png)

**Comment by PardeeCenterIFs** [comment]
Yes, the total value is calculated based on the sectoral values, and saved in our output file (working.run or .run.db), but what you show above is the historical analog used for display purposes, which doesn't necessarily match what we do in the preprocessor. This probably means that if the sum of the historical series of the sectors don't match this total series, then there will be a transient when doing history plus forecast for the total.
